cmd/finch/virtual_machine_status_test.go

```diff
@@ -0,0 +1,206 @@
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctrl := gomock.NewController(t)
+			logger := mocks.NewLogger(ctrl)
+			stdout := bytes.Buffer{}
+			lcc := mocks.NewLimaCmdCreator(ctrl)
+			lca := mocks.NewLimaConfigApplier(ctrl)
+
+			tc.mockSvc(lcc, logger, lca, ctrl)
+
+			assert.NoError(t, newStatusVMAction(lcc, logger, &stdout).runAdapter(tc.command, tc.args))
+		})
+	}
+}
+
+func TestStatusVMAction_run(t *testing.T) {
+	t.Parallel()
+
+	testCases := []struct {
+		name             string
+		wantErr          error
+		wantStatusOutput string
+		mockSvc          func(
+			*mocks.LimaCmdCreator,
+			*mocks.Logger,
+			*mocks.LimaConfigApplier,
+			*gomock.Controller,
+		)
+	}{
+		{
+			name:             "running VM",
+			wantErr:          nil,
+			wantStatusOutput: "Running\n",
+			mockSvc: func(
+				lcc *mocks.LimaCmdCreator,
+				logger *mocks.Logger,
+				lca *mocks.LimaConfigApplier,
+				ctrl *gomock.Controller,
+			) {
+				getVMStatusC := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("ls", "-f", "{{.Status}}", limaInstanceName).Return(getVMStatusC)
+				getVMStatusC.EXPECT().Output().Return([]byte("Running"), nil)
+				logger.EXPECT().Debugf("Status of virtual machine: %s", "Running")
+			},
+		},
+		{
+			name:             "stopped VM",
+			wantErr:          nil,
+			wantStatusOutput: "Stopped\n",
+			mockSvc: func(
+				lcc *mocks.LimaCmdCreator,
+				logger *mocks.Logger,
+				lca *mocks.LimaConfigApplier,
+				ctrl *gomock.Controller,
+			) {
+				getVMStatusC := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("ls", "-f", "{{.Status}}", limaInstanceName).Return(getVMStatusC)
+				getVMStatusC.EXPECT().Output().Return([]byte("Stopped"), nil)
+				logger.EXPECT().Debugf("Status of virtual machine: %s", "Stopped")
+			},
+		},
+		{
+			name:             "nonExistent VM",
+			wantErr:          nil,
+			wantStatusOutput: "Nonexistent\n",
+			mockSvc: func(
+				lcc *mocks.LimaCmdCreator,
+				logger *mocks.Logger,
+				lca *mocks.LimaConfigApplier,
+				ctrl *gomock.Controller,
+			) {
+				getVMStatusC := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("ls", "-f", "{{.Status}}", limaInstanceName).Return(getVMStatusC)
+				getVMStatusC.EXPECT().Output().Return([]byte(""), nil)
+				logger.EXPECT().Debugf("Status of virtual machine: %s", "")
+			},
+		},
+		{
+			name:             "undefined VM",
+			wantErr:          errors.New("unrecognized system status"),
+			wantStatusOutput: "",
+			mockSvc: func(
+				lcc *mocks.LimaCmdCreator,
+				logger *mocks.Logger,
+				lca *mocks.LimaConfigApplier,
+				ctrl *gomock.Controller,
+			) {
+				getVMStatusC := mocks.NewCommand(ctrl)
+				lcc.EXPECT().CreateWithoutStdio("ls", "-f", "{{.Status}}", limaInstanceName).Return(getVMStatusC)
+				getVMStatusC.EXPECT().Output().Return([]byte("Undefined"), nil)
+				logger.EXPECT().Debugf("Status of virtual machine: %s", "Undefined")
+			},
+		},
```

Comment:
something minor, @ningziwen cmiiw: re. https://github.com/runfinch/finch/pull/83#discussion_r1038437152, the "2 invalid cases" seem to refer to:

1. `cmd.Output()` inside `GetVMStatus` returns an error.
2. `toVMStatus` inside `GetVMStatus` returns an error.

This test case seems to cover the second point above, but that should be already covered by the `"unknown VM status"` test case below, so we may not need this test case.

```suggestion
```